tools/cloud-vision-utils/cloud_vision_utils/annotation.py

```diff
@@ -83,30 +83,35 @@ def read_from_pascal(filename: str):
   Args:
     filename: PASCAL VOC XML filename.
 
-  Yields:
-    `BoundingBox` objects.
+  Returns:
+    (image filename, list of `BoundingBox` objects).
   """
 
   with gfile.GFile(filename) as f:
     tree = ET.parse(f)
 
   root = tree.getroot()
+  image_filename = root.find('filename').text
```

Comment:
can we be guaranteed this will be found?

Reply:
Image labeling tools that support the PASCAL VOC format should include the source image 'filename' in the output annotation XML file. If the 'filename' element is not found then this function will throw an exception.